gobblin-api/src/main/java/gobblin/configuration/SourceState.java

```diff
@@ -92,8 +93,12 @@ public SourceState(State properties, List<WorkUnitState> previousTaskStates) {
   public synchronized Extract createExtract(Extract.TableType type, String namespace, String table) {
     Extract extract = new Extract(this, type, namespace, table);
     while (extractSet.contains(extract)) {
```

Comment:
`extractSet` is used here to make sure that each `Extract` has a unique `id`. I need this to unblock the Lynda gobblin job testing.